test/Microsoft.ML.Tests/Scenarios/IrisPlantClassificationTests.cs

```diff
@@ -65,53 +77,20 @@ public void TrainAndPredictIrisModelTest()
             Assert.Equal(.8, prediction.PredictedLabels[1], 1);
             Assert.Equal(0, prediction.PredictedLabels[2], 2);
 
-            // Note: Testing against the same data set as a simple way to test evaluation.
-            // This isn't appropriate in real-world scenarios.
-            string testDataPath = GetDataPath("iris.txt");
-            var testData = new TextLoader(testDataPath).CreateFrom<IrisData>(useHeader: false);
-
-            var evaluator = new ClassificationEvaluator();
-            evaluator.OutputTopKAcc = 3;
-            ClassificationMetrics metrics = evaluator.Evaluate(model, testData);
+            // Evaluate the trained pipeline
+            var predicted = trainedModel.Transform(testData);
+            var metrics = mlContext.MulticlassClassification.Evaluate(predicted, topK:3);
 
             Assert.Equal(.98, metrics.AccuracyMacro);
             Assert.Equal(.98, metrics.AccuracyMicro, 2);
-            Assert.Equal(.06, metrics.LogLoss, 2);
+            Assert.InRange(metrics.LogLoss, .05, .06);
```

Comment:
Could we use Assert.Equal(…, …, 2)?

Reply:
I will also prefer Equal

Reply:
fixed

---
In reply to: [245133023](https://github.com/dotnet/machinelearning/pull/2008#discussion_r245133023) [](ancestors = 245133023)